target/scripts/check-for-changes.sh

```diff
@@ -28,217 +28,255 @@ fi
 
 # ? --------------------------------------------- Actual script begins
 
-# determine postmaster address, duplicated from start-mailserver.sh
+# determine postmaster address
 # this script previously didn't work when POSTMASTER_ADDRESS was empty
-if [[ -n ${OVERRIDE_HOSTNAME} ]]
+# used in helper-functions.sh/_monitored_files_checksums
+if [[ -n "${OVERRIDE_HOSTNAME}" ]]
 then
-  DOMAINNAME="${OVERRIDE_HOSTNAME#*.}"
```

Comment:
Maybe I don't understand this, but the old DOMAINNAME we're creating here is wrong:
```
cat docker-mailserver/docker-compose.yml
version: '3.8'

services:
  mailserver:
    build:
      context: .
    hostname: xkr.email
    domainname: xkr.email
    container_name: deliver
    env_file: mailserver.env
```

```
head mailserver.env
# –––––––––––––––––––––––––––––––––––––––––––––––
# ––– Mailserver Environment Variables ––––––––––
# –––––––––––––––––––––––––––––––––––––––––––––––

# empty => uses the `hostname` command to get the mail server's canonical hostname
# => Specify a fully-qualified domainname to serve mail for.  This is used for many of the config features so if you can't set your hostname (e.g. you're in a container platform that doesn't let you) specify it in this environment variable.
OVERRIDE_HOSTNAME=xkr.email
```

```
HOSTNAME: xkr.email
DOMAINNAME: email
```